pandas/core/generic.py

```diff
@@ -9869,6 +9869,8 @@ def _where(
         # align the cond to same shape as myself
         cond = common.apply_if_callable(cond, self)
         if isinstance(cond, NDFrame):
+            # GH #52955: if cond is NA, element propagates in mask and where
+            cond = cond.fillna(True)
```

Comment:
has the option of just raising on NAs been discussed?  seems ambiguous and a general PITA.

Reply:
i would just have that raise too, yes.

Reply:
@topper-123 I think we may need further discussion about the desired behavior of `_where`.

Reply:
IMO we should accept  BooleanArrays (and Series/DataFrame containing BooleanArrays/ArrowArray[bool]) as conditional here. I think it will be surprising if those data structure work in `loc` and not here.

Do similar functionality raise in any other methods? I don't recall any.

Reply:
Hi @jbrockmendel any updates on this?